### PR TITLE
Story 13

### PR DESCRIPTION
**Pull request by mfaisalnazari**
finished with tests i believe we would have to meet and merge this all.

### CODE REVIEW

File: app/models/merchant.rb [hunk 1]
```diff
@@ -33,4 +33,33 @@ def items_ready_to_ship
...
+        .joins(:invoice_items)
+        .joins(invoices: :transactions) # Change 'invoice' to 'invoices'
+        .where(transactions: { result: 'success' })
+        .group('items.id')
+        .order('total_revenue DESC')
+        .limit(5)
+  end
+
+  def calculate_topitems_date(top_items)
+    top_items.map do |item|
```
Comment (odellmac4) [inline app/models/merchant.rb:49]: @mfaisalnazari be careful her because I see the `map` method and think that will be counted as using ruby

Comment (odellmac4) [reply app/models/merchant.rb:49]: what are you trying to do by iterating?

if you're trying to access a merchant's items, it  already has_many items association. Just need clarification why using `map`

Comment (mfaisalnazari) [reply app/models/merchant.rb:49]: cuz there is 5 items i have to do that for all of five of them basically iterate for each and the look into the database for the query related to that item
maybe i can merge both methods together but i think every way you'd have to first pull the top item one by one and then find the top selling date for that item

Comment (hoadam) [reply app/models/merchant.rb:49]: I understand what Faisal r doing but I also have the same question as Odell's. I personally think it is ok and I plan to do the same thing for best_day of top_merchant but I will use map on the view.

**Pull request closed without merging**